File: src/XamarinNativeAndroidDemo/MovieDetailActivity.cs
using Android.App;
using Android.OS;
using Android.Widget;
using AndroidDemo.Model;
using Newtonsoft.Json;

namespace AndroidDemo
{
    [Activity(Label = "@string/MovieDetailText", Icon = "@drawable/icon", Theme = "@style/Theme.Custom")]
    public class MovieDetailActivity : Activity
    {
        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);
            SetContentView(Resource.Layout.MovieDetails);
            var movie =JsonConvert.DeserializeObject<Movie>(Intent.GetStringExtra("movie"));
            BindData( movie);
        }

        public  void BindData(Movie movie)
        {
            var name = FindViewById<TextView>(Resource.Id.txtName);
            var grade = FindViewById<TextView>(Resource.Id.txtGrade);
            var storyType = FindViewById<TextView>(Resource.Id.txtStory);
            var director = FindViewById<TextView>(Resource.Id.txtDirector);
            var actor = FindViewById<TextView>(Resource.Id.txtActor);

            name.Text = movie.Name;
            grade.Text = movie.Grade;
            storyType.Text = movie.Story;
            director.Text = movie.Director;
            actor.Text = movie.Actor;
        }
    }
}

File: src/RetrofitDemo/app/src/main/java/com/retrofitdemo/michael/retrofitdemo/MovieDetailActivity.java
package com.retrofitdemo.michael.retrofitdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;
import com.google.gson.Gson;
import com.retrofitdemo.michael.retrofitdemo.model.Movie;

public class MovieDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movie_detail);

        Movie movie =new Gson().fromJson(getIntent().getStringExtra("movie"),Movie.class);

        BindData(movie);
    }

    public  void BindData(Movie movie)
    {
        TextView name = (TextView)findViewById(R.id.txtName);
        TextView grade = (TextView)findViewById(R.id.txtGrade);
        TextView storyType =(TextView)findViewById(R.id.txtStory);
        TextView director = (TextView)findViewById(R.id.txtDirector);
        TextView actor = (TextView)findViewById(R.id.txtActor);

        name.setText(movie.getName());
        grade.setText(movie.getGrade());
        storyType.setText(movie.getStory());
        director.setText(movie.getDirector());
        actor.setText(movie.getActor());
    }
}
